lib/syscall_shim/include/uk/syscall.h

```diff
@@ -44,14 +44,53 @@
 #include <uk/errptr.h>
 #include <errno.h>
 #include <stdarg.h>
+#include <stddef.h>
 #include <uk/print.h>
 #include <uk/legacy_syscall.h>
 #include <uk/bits/syscall_linuxabi.h>
+#include <uk/syscall_exittab.h>
+#include <uk/syscall_entertab.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+static inline void uk_syscall_enter_handler(struct uk_syscall_enter_ctx *sctx)
+{
+	/* Mark system call entry */
+	uk_syscall_nested_depth++;
+
+#if CONFIG_LIBSYSCALL_SHIM_ASSERT_NESTED_SYSCALLS
+	if (unlikely(uk_syscall_nested_depth != 1))
+		UK_CRASH("Nested system call detected. Depth: 0x%02x\n",
+			 uk_syscall_nested_depth);
+#else /* !CONFIG_LIBSYSCALL_SHIM_ASSERT_NESTED_SYSCALLS */
+	/* Only run this if the first system call in the callchain */
```

Comment:
We should provide some reasoning on why we do that, in a comment or in the commit message (or both).